### PR TITLE
refactor(manager/kustomize): refactor Kustomize interface

### PR DESCRIPTION
**Pull request by andrein**
## Changes:

Refactor kustomize interface based on feedback from #12628 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/kustomize/types.ts [hunk 1]
```diff
@@ -6,6 +6,6 @@ export interface Image {
 }
 export interface Kustomize {
   kind: string;
-  bases: string[];
-  images: Image[];
```
Comment (viceice) [inline lib/manager/kustomize/types.ts:10]: leave those legacy types.

File: lib/manager/kustomize/extract.spec.ts [hunk 2]
```diff
@@ -244,7 +244,6 @@ describe('manager/kustomize/extract', () => {
       expect(res).not.toBeNull();
       expect(res.deps).toMatchSnapshot();
       expect(res.deps).toHaveLength(3);
-      expect(res.deps[0].currentValue).toBe('v0.0.1');
```
Comment (viceice) [inline lib/manager/kustomize/extract.spec.ts:247]: Why is this removed?

File: lib/manager/kustomize/extract.ts [hunk 3]
```diff
@@ -122,11 +122,10 @@ export function parseKustomize(content: string): Kustomize | null {
     return null;
   }
 
-  pkg.bases = (pkg.bases || []).concat(
-    pkg.resources || [],
-    pkg.components || []
+  pkg.resources = (pkg.resources ?? []).concat(
+    pkg.bases ?? [], // bases is deprecated since Kustomize v2.1.0
+    pkg.components ?? []
   );
```
Comment (viceice) [inline lib/manager/kustomize/extract.ts:128]: i like to completly avoid this manipulation here and simply add multiple `for ... of ...` loops to extract function.

File: lib/manager/kustomize/extract.spec.ts (hunk 2 above)
Comment (andrein) [reply lib/manager/kustomize/extract.spec.ts:247]: Typo, fixed

File: lib/manager/kustomize/extract.ts [hunk 4]
```diff
@@ -107,7 +107,7 @@ export function extractImage(image: Image): PackageDependency | null {
 }
 
 export function parseKustomize(content: string): Kustomize | null {
-  let pkg = null;
+  let pkg: Kustomize | null | undefined = null;
```
Comment (viceice) [inline lib/manager/kustomize/extract.ts:110]:
```suggestion
  let pkg: Kustomize | null = null;
```

I think `load` will never return `undefined`

**Review by viceice — DISMISSED** [review]

**Review by viceice — APPROVED** [review]
